### PR TITLE
Update KOReader to 2021.04

### PR DESCRIPTION
**Pull request by Eeems**
Also change maintainer to Eeems

List of remarkable specific issues resolved in this release:

- https://github.com/koreader/koreader/issues/7519 'suspend', 'reboot' and 'poweroff' routines
  - https://github.com/koreader/koreader/pull/7563 remove extra refresh before poweroff
  - https://github.com/koreader/koreader/issues/7139 reMarkable 2 wakeup timing
- https://github.com/koreader/koreader/pull/7536 Fix for https://github.com/koreader/koreader/pull/7415 breaking reMarkable touch input
- https://github.com/koreader/koreader/issues/7508 Remarkable: "sync" command replaced as of FW 2.6
  - https://github.com/koreader/koreader/pull/7509 remarkable FW 2.6: Set absolute path for 'sync'

### CODE REVIEW

**Comment by Eeems** [comment]
Quick test on rM1 shows it installs fine, opens, touch/pen work. Network control seems to work as well, although it's not playing nice with Oxide, as the network list is empty in Oxide after using KOReader to connect.

**Comment by raisjn** [comment]
cc @snelg

**Comment by snelg** [comment]
I don't use toltec, but I'm willing to help test stuff. So I installed toltec on my rm2, switched to the `testing` configuration, and then installed Oxide. Not sure where things went wrong, but I couldn't get Oxide to work at all. I'm on firmware 2.6, so maybe that's the problem?

**Comment by raisjn** [comment]
the version of qt in 2.6 is incompatible with the toolchains, but there is a toltec image for qt that can be used to build oxide. this qt issue has been holding up toltecs official release and support of rM2 :(

you can also try remux (no dep on qt)

**Comment by raisjn** [comment]
cc @witosso for oxide builds on 2.6

**Comment by snelg** [comment]
Same basic results as Eeems. WiFi in KOReader works a little different under the hood, so live-switching between KOReader and Xochitl had a little bit of strangeness.
They both use `wpa_supplicant`, but Xochitl launches it with the `-u` flag for DBus interface, while KOReader launches it with the flags `-i wlan0 -C /var/run/wpa_supplicant -B` so it can use `wpa_cli` to interface.

Aaaand then I effectively bricked my rm2 :( :( :(
My own dang fault; step-by-step manual cleaning out all the toltec packages seemed like a hassle, and I have my own KOReader stuff backed up. So I used the "Factory Reset" option in Xochitl.
That was a mistake. I'm not sure what the Factory Reset cleans up and what it doesn't, but
 1) it's stuck on the "reMarkable is starting" screen
 2) it definitely changed the root password; USB ssh port works, but password is rejected

Sigh. Time to get some hardware, I guess. https://github.com/ddvk/remarkable2-recovery

**Comment by Eeems** [comment]
> I'm not sure what the Factory Reset cleans up and what it doesn't, but

From my understanding, just your user data. Which means if you've switched to having something other than xochitl enabled it will break things.

**Comment by snelg** [comment]
> Which means if you've switched to having something other than xochitl enabled it will break things.

Yeah, I figured that part out :)

File: package/koreader/package
Comment (raisjn) [inline package/koreader/package]: i've thought about it a bit and i think i'll stay maintainer. the maintainer's job is to accept changes / review PRs which i think i've been doing. if you become maintainer and then put up a PRs, i think the process will be slower than if i am maintainer

**Review by raisjn — CHANGES_REQUESTED** [review]
> Aaaand then I effectively bricked my rm2 :( :( :(

i'm really sorry! if you go on discord, i'm sure bokluk and others will help you recover when hardware arrives

> They both use wpa_supplicant, but Xochitl launches it with the -u flag for DBus interface, while KOReader launches it with the flags -i wlan0 -C /var/run/wpa_supplicant -B so it can use wpa_cli to interface.

is this a blocker?

**Comment by Eeems** [comment]
> > They both use wpa_supplicant, but Xochitl launches it with the -u flag for DBus interface, while KOReader launches it with the flags -i wlan0 -C /var/run/wpa_supplicant -B so it can use wpa_cli to interface.
> 
> is this a blocker?

I don't think so. It's a valid way to try to handle wifi on the device. I do think we should open an issue or PR for KOReader to have it handle wifi by just using the standard DBus interface to play nicer with Xochitl and Oxide though.

**Comment by snelg** [comment]
> > Aaaand then I effectively bricked my rm2 :( :( :(
> 
> i'm really sorry! if you go on discord, i'm sure bokluk and others will help you recover when hardware arrives

Thanks for the sympathy :)
I hit a few snags along the way (bad soldering, accidentally using a USB-C cable only wired for USB 2.0, etc.), but managed to work through them. My rm2 is working fine again.

**Review by raisjn — APPROVED** [review]

**Pull request merged**
